src/main/InternalProcess.js

```diff
@@ -0,0 +1,150 @@
+import { fork } from "child_process";
+import logger from "~/logger";
+
+class InternalProcess {
+  constructor(timeout) {
+    this.process = null;
+
+    this.timeout = timeout;
+    this.active = false;
+
+    this.onStartCallback = null;
+    this.onMessageCallback = null;
+    this.onExitCallback = null;
+
+    this.messageQueue = [];
+    this.sending = false;
+  }
+
+  onMessage(callback) {
+    this.onMessageCallback = callback;
+  }
+
+  onExit(callback) {
+    this.onExitCallback = callback;
+  }
+
+  run() {
+    this.sending = true;
+    while (this.messageQueue.length && this.active) {
+      const message = this.messageQueue.shift();
+      this.process.send(message);
+    }
+    this.sending = false;
+  }
+
+  send(message) {
+    this.messageQueue.push(message);
```

Comment:
so every send is doing a push and instantly pushing the messages. why not doing `this.process.send(message)` directly? it seems we don't really need this messageQueue system.

Reply:
It is not needed most of the time but because restarting the internal process is asynchrone and can take up to two seconds, this ensure that no message are lost in the meantime.